src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -273,16 +277,36 @@ def csv(self, path: str) -> DataFrame:
 
         self._file_path = path
         self._file_type = "csv"
-        df = DataFrame(
-            self._session,
-            self._session._plan_builder.read_file(
-                path,
-                self._file_type,
-                self._cur_options,
-                self._session.get_fully_qualified_current_schema(),
-                self._user_schema._to_attributes(),
-            ),
-        )
+        from snowflake.snowpark import context
+
+        if context._use_sql_simplifier:
+            df = DataFrame(
+                self._session,
+                SelectStatement(
+                    from_=SelectSnowflakePlan(
+                        self._session._plan_builder.read_file(
+                            path,
+                            self._file_type,
+                            self._cur_options,
+                            self._session.get_fully_qualified_current_schema(),
+                            self._user_schema._to_attributes(),
+                        ),
+                        analyzer=self._session._analyzer,
+                    ),
+                    analyzer=self._session._analyzer,
```

Comment:
This may be out of scope for this PR, but I noticed that we are passing analyzer twice several times. Can we update code so that `SelectStatement` uses analyzer from `from_` whenever available?

Reply:
Yeah that's doable. `SelectStatement` can grab the session from `from_`.